src/charm_state.py

```diff
@@ -372,6 +377,33 @@ def _parse_openstack_clouds_config(cls, charm: CharmBase) -> dict | None:
 
         return cast(dict, openstack_clouds_yaml)
 
+    @validator("reconcile_interval")
+    @classmethod
+    def check_reconcile_interval(cls, reconcile_interval: int) -> int:
+        """Validate the general charm configuration.
+
+        Args:
+            reconcile_interval: The value of reconcile_interval passed to class instantiation.
+
+        Raises:
+            ValueError: if an invalid reconcile_interval value of less than 2 has been passed.
+
+        Returns:
+            The validated reconcile_interval value.
+        """
+        # The EventTimer class sets a timeout of `reconcile_interval` - 1.
```

Comment:
This is no longer true anymore, due to using of `run-one` now.
This can be updated in a follow-up PR.